### PR TITLE
Copy backup file during dump agent.toml

### PR DESCRIPTION
**Pull request by jil8885**
This pull reqeust resolves https://github.com/lablup/backend.ai/issues/349
Copy agent,toml to agent.toml.bak and dump new config into agent.toml

### CODE REVIEW

File: src/ai/backend/agent/server.py [hunk 1]
```diff
@@ -292,6 +294,7 @@ async def update_scaling_group(self, scaling_group):
             data = tomlkit.load(f)
             data['agent']['scaling-group'] = scaling_group
         with open(cfg_src_path, 'w') as f:
+            shutil.copy(cfg_src_path, f"{cfg_src_path}.bak")
```
Comment (achimnol) [inline src/ai/backend/agent/server.py:297]: It is kind-of safer to copy the file before opening it.
```suggestion
        shutil.copy(cfg_src_path, f"{cfg_src_path}.bak")
        with open(cfg_src_path, 'w') as f:
```

And let's wrap the method body using [an executor](https://docs.python.org/3.10/library/asyncio-eventloop.html#executing-code-in-thread-or-process-pools) and `asyncio.Lock` to keep it async and prevent potential races on updates.

File: changes/339.feature [hunk 2]
```diff
@@ -0,0 +1 @@
+Copy agent.toml to agent.toml.bak before dump new config into agent.toml
```
Comment (achimnol) [inline changes/339.feature:1]:
```suggestion
Copy `agent.toml` to `agent.toml.bak` before modifying it to update the scaling group
```

**Review by achimnol — CHANGES_REQUESTED** [review]

File: src/ai/backend/agent/server.py [hunk 3]
```diff
@@ -1,5 +1,7 @@
 from __future__ import annotations
 
+import shutil
+
 import tomlkit
```
Comment (achimnol) [inline src/ai/backend/agent/server.py:5]: Please move these imports to follow the alphabetical order and distinguishable sections of stdlib, 3rd party libs, other pkgs under the same `ai.backend` namespace, and internal relative imports.

Comment (jil8885) [reply src/ai/backend/agent/server.py]: Resolved at https://github.com/lablup/backend.ai-agent/pull/339/commits/cb2ee1c487dc952debe583c95069ec8f3668f0d5

File: src/ai/backend/agent/server.py [hunk 4]
```diff
@@ -35,6 +34,8 @@
 from callosum.lower.zeromq import ZeroMQAddress, ZeroMQRPCTransport
 import click
 from setproctitle import setproctitle
+import shutil
```
Comment (achimnol) [inline src/ai/backend/agent/server.py:37]: T_T
Since `shutil` is a stdlib, it should be placed on the upper block, before `import signal`.

**Review by achimnol — CHANGES_REQUESTED** [review]

File: src/ai/backend/agent/server.py (hunk 4 above)
Comment (jil8885) [reply src/ai/backend/agent/server.py]: I forgot it is stdlib, so I rearrange it with other stdlib at https://github.com/lablup/backend.ai-agent/pull/339/commits/6435abd7e7a36c9afc3ea70c5b7a27f9c0f497cd

**Pull request merged**